### DISCUSSION TITLE
Should Unit-Testing for Vue components still be supported?

### DISCUSSION THREAD

**Discussion by devtobi** [it-at-m/refarch-templates#127] [General]
There was not that much consent for unit testing Vue components in the past. We should discuss if we want to further include this.

**Comment by eidottermihi** [comment]
I recently tried Cypress for "Component Testing" and I personally liked it a lot more than Vitest + vue-test-utils because you could actually "see" the component under test and therefore "test development" and debugging problems was a lot easier.

https://docs.cypress.io/guides/component-testing/overview

**Reply by devtobi** [reply]
Technically a DOM implementation like jsdom would be enough to run unit tests, but we didn't think about the DevEx as you mentioned. Thanks for that point of view.
Vitest has kind of the same feature (but still experimental). See https://vitest.dev/guide/browser/. The additional required https://github.com/vitest-dev/vitest-browser-vue is also it a very very early stage.
I think we should compare those two solutions and then decide for one long term. I only knew Cypress as E2E tests tool and didn't know about this feature.
However I think we use a tool that also allows headless operation, so the developer can decide if he wants to see the UI or not. ;)

**Reply by eidottermihi** [reply]
> However I think we use a tool that also allows headless operation

Cypress component testing doesn't enforce a UI, you can run the tests headless as well (like in CI)
